### ISSUE TITLE
symbols.json file structure

### ISSUE THREAD

**Issue by dvergeylen** [daniel3735928559/guppy#132]
Hi Daniel,

Playing a bit with the `symbols.json` file, I was wondering why it was simply a hash of symbols, as this doesn't allow to specify an order of symbols (neither of tabs).

This might be a bit confusing at first but why couldn't we define the symbols.json file as array of groups, containing an array of symbols? Something like this:

```json
{
    "_version": "2.0.0-alpha.3",
    "_name": "base",
    "groups": [
        { "name": "functions",
          "symbols": [
          {  "name":  "symbol1",
              "attrs": "apple"
           },
           { "name": "symbol2",
              "attrs": "pear"
           }
        ]},
      { "name": "operators",
        "symbols": []
      }
    ]
}
```

Parsing arrays sequentially, 

* It would guarantee an order for the symbols and the tabs on the OSK
* It would allow not to hardcode `group_headers` as they could be derived from a `.reduce` on the json file
* It would allow to convert "abc", "ABC" and "arith" tabs into the `symbols.json` files. As a french speaker my keyboard is Azerty and not Qwerty for instance and since these tabs are hardcoded I can't simply tune the `symbols.json` file to localize the keyboard.

I can deliver a PR for this, as I will implement this in my fork for personal purpose anyway :wink:

**Comment by saivan** [comment]
Yes I agree with this too. This makes a lot of sense, and would also help to solve the problem encountered in #91, as you could avoid overwriting symbols if an earlier symbol also matches the current text. Javascript objects *dont need* to keep order in tact, although as you rightly point out, on most browsers they currently do :P
